engine/access/rest/websockets/data_providers/blocks_provider_test.go

```diff
@@ -0,0 +1,120 @@
+package data_providers
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/suite"
+
+	accessmock "github.com/onflow/flow-go/access/mock"
+	"github.com/onflow/flow-go/engine/access/rest/common/parser"
+	"github.com/onflow/flow-go/engine/access/rest/websockets/models"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+const unknownBlockStatus = "unknown_block_status"
+
+type testErrType struct {
+	name             string
+	arguments        models.Arguments
+	expectedErrorMsg string
+}
+
+// BlocksProviderSuite is a test suite for testing the block providers functionality.
+type BlocksProviderSuite struct {
+	suite.Suite
+
+	log zerolog.Logger
+	api *accessmock.API
+
+	blockMap       map[uint64]*flow.Block
+	rootBlock      flow.Block
+	finalizedBlock *flow.Header
+}
+
+func TestBlocksProviderSuite(t *testing.T) {
```

Comment:
can you also include happy path tests?

Reply:
[f8d711f](https://github.com/onflow/flow-go/pull/6636/commits/f8d711f75a9219c44f4b9831f5c713a80c9d28bc)